server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSessionServlet.scala

```diff
@@ -194,10 +205,11 @@ class InteractiveSessionServlet(livyConf: LivyConf)
   }
 
   jpost[AddResource]("/:id/add-jar") { req =>
-    withSession { lsession =>
-      val uri = new URI(req.uri)
-      lsession.addJar(uri)
-    }
+    addJarOrPyFile(req)
+  }
+
+  jpost[AddResource]("/:id/add-pyfile") { req =>
+    addJarOrPyFile(req)
```

Comment:
It would be nice to check that this is indeed a python session before trying to even make a remote call.